surveys/views.py

```diff
@@ -15,21 +16,82 @@
 
 from .models import SurveyFormEntry, SurveyChart
 from .forms import StudyCreateForm, StudyAreaCreateForm, SurveyCreateForm, \
-                   SurveyChartForm
+                   SurveyChartForm, LocationCreateForm, LocationAreaCreateForm, \
+                   LocationLineCreateForm, AgencyCreateForm
 
 
 class AgencyCreate(CreateView):
+    form_class = AgencyCreateForm
     model = Agency
     template_name = "agency_create.html"
-    fields = '__all__'
     success_url = '/'
 
 
 class LocationCreate(CreateView):
-    model = Location
+    form_class = LocationCreateForm
+    form_class_location_area = LocationAreaCreateForm
+    form_class_location_line = LocationLineCreateForm
+
     template_name = "location_create.html"
-    fields = '__all__'
-    success_url = '/'
+    success_url = reverse_lazy('surveys-create')
+
+    def get_context_data(self, **kwargs):
+        context = super(LocationCreate, self).get_context_data(**kwargs)
+
+        context['form_location'] = self.form_class(prefix="location")
+        context['form_location_area'] = self.form_class_location_area(prefix="location-area")
+        context['form_location_line'] = self.form_class_location_line(prefix="location-line")
```

Comment:
When we generate forms, can we use `initial` to set the default country to the US?

Reply:
Nice idea!

Reply:
I'm also setting the initial agency value to `Agency.objects.first()`—I think the most common use case will be that a user has permission to make a location for only one agency.